internal/discord/commands.go

```diff
@@ -1,11 +1,36 @@
 package discord
 
 import (
+	"fmt"
+
 	"github.com/bwmarrin/discordgo"
 )
 
 func (d *Discord) Kick(s *discordgo.Session, i *discordgo.InteractionCreate) {
-
+	options := i.ApplicationCommandData().Options
+	optionMap := make(map[string]*discordgo.ApplicationCommandInteractionDataOption, len(options))
+	for _, opt := range options {
+		optionMap[opt.Name] = opt
+	}
+
+	userToKick := optionMap["user"].UserValue(nil).ID
+
+	var err error
+	if optionMap["reason"] != nil {
+		err = d.Session.GuildMemberDeleteWithReason(i.GuildID, userToKick, optionMap["reason"].StringValue())
+	} else {
+		err = d.Session.GuildMemberDelete(i.GuildID, userToKick)
+	}
```

Comment:
we don't want to kick a user without a specific reason, so we should check that there is a reason. if there is not, we yell at the mod to give us a reason before kicking